File: pcpartsbackend/src/main/java/erp/pcpartsbackend/models/Product.java
package erp.pcpartsbackend.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.io.Serializable;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="products")
public class Product implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID productId;
    private String productName;
    private String productCategory;
    private Float productPrice;
    private int quantityInStock;
    private String imageUrl;
    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "providerId")
    private Provider provider;
}

File: pcpartsfrontend/pcpartsshop/src/app/models/Product.ts
import { Provider } from './Provider';

export class Product {
  constructor(
    public productId: string,
    public productName: string,
    public productCategory: string,
    public productPrice: number,
    public quantityInStock: number,
    public imageUrl: string,
    public providerId: string, // Use Provider object instead of providerId
  ) {}
}
